packages/ui/origin-ui-core/src/components/certificates/CertificatesInbox.tsx

```diff
@@ -0,0 +1,456 @@
+            fontSize: '14px',
+            color: TEXT_COLOR_DEFAULT
+        }
+    });
+
+    const classes = useStyles();
+
+    const setEnergyForCertificate = (
+        device: IInboxItemData,
+        cert: IInboxCertificateData,
+        value: BigNumber
+    ) => {
+        const newViewData = viewData.map((d) => {
+            if (d.id === device.id) {
+                d.certificates = d.certificates.map((c) => {
+                    if (c.id === cert.id) {
+                        return {
+                            ...c,
+                            energy: value
+                        };
+                    }
+                    return c;
+                });
+            }
+            return d;
+        });
+
+        setViewData(newViewData);
+    };
+
+    async function publishForSale() {
+        const certs = getSelectedCertificates();
+
+        certs.forEach((certificate) => {
+            dispatch(
+                requestPublishForSale({
+                    certificateId: parseInt(certificate.id, 10),
+                    amount: certificate.energy,
+                    price: Math.round((price + Number.EPSILON) * 100),
+                    callback: () => {
+                        updateView();
+                    },
+                    source: certificate.source,
+                    assetId: certificate.assetId
+                })
+            );
+        });
+    }
+
+    async function withdraw() {
+        const certs = getSelectedCertificates();
+        const address = user.blockchainAccountAddress;
+
+        certs.forEach((certificate) => {
+            const assetId = certificate.assetId;
+            const amount = certificate.energy.toString();
+            dispatch(
+                requestWithdrawCertificate({
+                    assetId,
+                    address,
+                    amount,
+                    callback: () => {
+                        updateView();
+                    }
+                })
+            );
+        });
+    }
+
+    const SelectedCertificatesList = (): [IInboxItemData, IInboxCertificateData][] => {
+        const selectedDeviceCertPairs: [IInboxItemData, IInboxCertificateData][] = [];
+
+        viewData.forEach((d) => {
+            d.certificates.forEach((c) => {
+                if (selectedCerts.includes(c.id)) {
+                    selectedDeviceCertPairs.push([d, c]);
+                }
+            });
+        });
+
+        return selectedDeviceCertPairs;
+    };
+
+    return (
+        <div>
+            <InboxItemPreview
+                open={modalData !== null}
+                setOpen={() => setModalData(null)}
+                data={modalData}
+            />
+            <Grid container spacing={3}>
+                <Grid item xs={12} md={7}>
+                    <div className={classes.box}>
+                        <Typography className={classes.header}>
+                            {t('certificate.info.certificatesToImport')}
```

Comment:
I guess this is not a `Certificates Import`. 
Title should be either `Certificates` of `Exchange Inbox`.
Also you have update the component name to `ExchangeInbox`, but the file name is still `CertificatesInbox`

Reply:
file renamed, title changed
bugs with total volume and selecting checkboxes patched
routing bug seems to be fixed (didn't appear for me after reordering routes in menuCreator)